### DISCUSSION TITLE
Dependency Injection in e2e testing

### DISCUSSION THREAD

**Discussion by MateusRodriguesCosta** [cypress-io/cypress#22496] [Questions and Help]
Hi folks,

We have a situation here where our component has some injected dependencies via a constructor that is used inside our HTML file.

The service injected comes from an external library inside our modules.

How could we stub the object there? Since the instance is already in place, we cannot refer directly to Its methods (only the static ones).

I was wondering if It has anything like Jasmine where we can do something like "TestBed.Inject(ServiceClass)" and then use the result as the object to be passed in the stub method.

![sbil-feature-toggle-view – feature-toggle spec ts](https://user-images.githubusercontent.com/6730901/175398475-400120f4-f23c-45ec-b619-255c7168ff4f.jpg)

Thanks guys.

**Comment by MateusRodriguesCosta** [comment]
Is there any news guys?

**Comment by muhamedkarajic** [comment]
I'm also interested in this.
